.github/check_doxygen.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python3
+
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+import os
+import re
+import subprocess
+import sys
+from pathlib import Path
+
+TMP_FOLDER = "/tmp/doxygen_check"
+WARN_FILE_NAME = "warnings.log"
+
+
+def run_cmd(cmd: str) -> list[str]:
+    out = subprocess.check_output(cmd, text=True, shell=True)
+    return out.splitlines()
+
+
+def run_cmd_no_check(cmd: str) -> list[str]:
+    out = subprocess.run(cmd, text=True, shell=True, stdout=subprocess.PIPE,
+                         stderr=subprocess.DEVNULL).stdout
+    return out.splitlines()
+
+
+def is_ignored(filename: str) -> bool:
+    if Path(filename).suffix != ".h" or "priv" in filename:
+        return True
+    if filename == "nimble/include/nimble/ble.h":
+        return False
```

Comment:
Why this particular file is about to be ignored?

Consider having a list of ignored files that can be easily extended in the future.
```
ignored_files = [
   "nimble/include/nimble/ble.h",
]

(...)

if (filename in ignored_files):
  return False
```

Reply:
Thanks, will add the list.

Actually the mentioned file is about to be included (is_ignored returns False), because it is outside the directory to be included which is specified by the regular expression. Nonetheless, it contains code widely used throughout the NimBLE, so it is desirable to have it documented.